src/api/clients/HttpClient.ts

```diff
@@ -0,0 +1,14 @@
+type Url = string;
+type Config = { params: URLSearchParams };
+
+export type HttpClient = {
+  get<D = unknown>(
+    url: Url,
+    config?: Partial<Config & D>,
+  ): Promise<{ data: D }>;
+  post<D = unknown, B = unknown>(
+    url: Url,
+    body?: B,
+    config?: Partial<Config & D & B>,
+  ): Promise<{ data: D }>;
+};
```

Comment:
```suggestion
export type HttpClient = {
  get<D = unknown>(
    url: Url,
    config?: Partial<Config>,
  ): Promise<{ data: D }>;
  post<D = unknown, B = unknown>(
    url: Url,
    body?: B,
    config?: Partial<Config>,
  ): Promise<{ data: D }>;
};
```
이렇게 정석처럼 짜면 lint error 발생함.

참고) 발생하는 lint error

> Type parameter D is used only once in the function signature.eslint[@typescript-eslint/no-unnecessary-type-parameters](https://typescript-eslint.io/rules/no-unnecessary-type-parameters)

Reply:
아래처럼 쓰지 않으면 린트 에러가 나는건가요? config에 저게 왜 들어가나 했습니다😵‍💫

Reply:
넵 맞습니다. 더 좋은 해법이 있을 것 같아 고민해보겠습니다.